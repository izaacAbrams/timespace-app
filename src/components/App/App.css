.header {
  margin: 0;
  padding: 20px;
  font-family: "Fjalla One", sans-serif;
  font-weight: 100;
  font-size: 32px;
  text-align: center;
  color: #edf2f4;
}

body {
  background-color: #151515;
}

.red {
  font-size: 16px;
  color: red;
}

.input_label {
  font-family: "Roboto", sans-serif;
  font-size: 20px;
}

.modal {
  display: none;
  position: fixed;
  z-index: 1;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgb(0, 0, 0);
  background-color: rgba(0, 0, 0, 0.4);
}

.display-modal {
  display: flex;
  flex-direction: column;
  background-color: #edf2f4;
  border-radius: 5px;
  width: 50%;
  margin: 15% auto;
  text-align: center;
  padding: 20px;
}

.modal-close {
  align-self: flex-end;
  color: #161925;
  margin: 0;
  font-size: 18px;
}

@media screen and (max-width: 510px) {
  .Appointments__main_appt {
    font-size: 14px;
  }
  .Appointments__notes {
    font-size: 12px;
  }
  .ApptCard__btn {
    padding: 3px;
  }
  .LandingPage__title {
    font-size: 35px;
  }
  .date-card-date {
    font-size: 15px;
  }
  .LandingPage__button {
    font-size: 12px;
    padding: 5px;
  }
  .LandingPage {
    padding: 20% 10%;
  }
}
