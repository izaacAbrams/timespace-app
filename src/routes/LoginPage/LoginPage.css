.LoginPage__section {
  background-image: url("../../images/login_page.svg");
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  height: calc(100vh - 59px);
}

.LoginPage__header {
  margin: 0;
  padding: 20px;
  font-family: "Fjalla One", sans-serif;
  text-align: center;
  color: #edf2f4;
}
