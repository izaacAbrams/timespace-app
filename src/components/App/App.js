import React, { Component } from "react";
import { Route, Switch } from "react-router-dom";
import moment from "moment";
import Nav from "../Nav/Nav";
import LandingPage from "../../routes/LandingPage/LandingPage";
import SignUpForm from "../SignUpForm/SignUpForm";
import NewApptForm from "../NewApptForm/NewApptForm";
import Appointments from "../../routes/Appointments/Appointments";
import LoginPage from "../../routes/LoginPage/LoginPage";
import NewScheduleForm from "../../routes/NewScheduleForm/NewScheduleForm";
import Schedules from "../../routes/Schedules/Schedules";
import ApptApiService from "../../services/appt-api-service";
import ScheduleApiService from "../../services/schedule-api-service";
import PrivateRoute from "../Utils/PrivateRoute";
import NotFound from "../../routes/NotFound/NotFound";
import Success from "../../routes/Success/Success";
import TimespaceContext from "../../contexts/TimespaceContext";
import TokenService from "../../services/token-service";
import AuthApiService from "../../services/auth-api-service";
import IdleService from "../../services/idle-service";

import "./App.css";

class App extends Component {
  static contextType = TimespaceContext;

  state = {
    currentSchedule: {},
    apptList: [],
    scheduleList: [],
    apptTimesList: [],
    appt_modal: false,
    schedule_modal: false,
    selected_date: moment().format(),
    addAppt: (appt) => {
      ApptApiService.postAppt(appt);
      this.state.apptList.push(appt);
    },
    addApptList: (scheduleId) => {
      ApptApiService.getAppts(scheduleId)
        .then((apptList) => {
          this.setState({ apptList });
        })
        .catch(this.state.setError);
    },
    addScheduleList: (userId) => {
      ScheduleApiService.getSchedules(userId).then((scheduleList) => {
        this.setState({ scheduleList });
      });
    },
    addApptTimesList: (scheduleId) => {
      ApptApiService.getTimesForNewAppt(scheduleId).then((appt) => {
        const apptTimesList = appt.map((appt) => {
          return {
            appt_date_time: moment(appt.appt_date_time).format(),
          };
        });
        this.setState({ apptTimesList });
      });
    },
    changeSelectedDate: (date) => {
      this.setState({ selected_date: date });
    },
    deleteAppt: (apptId) => {
      ApptApiService.deleteAppt(apptId);
      this.setState({
        apptList: this.state.apptList.filter((appt) => apptId !== appt.id),
      });
    },
    deleteSchedule: (scheduleId) => {
      ScheduleApiService.deleteSchedule(scheduleId);
      this.setState({
        scheduleList: this.state.scheduleList.filter(
          (schedule) => scheduleId !== schedule.id
        ),
      });
    },
    patchAppt: (apptId, updatedAppt) => {
      ApptApiService.patchAppt(apptId, updatedAppt);
      const updatedApptList = this.state.apptList.map((appt) => {
        return appt.id === apptId ? updatedAppt : appt;
      });
      this.setState({
        apptList: updatedApptList,
      });
    },
    patchSchedule: (scheduleId, updatedSchedule) => {
      ScheduleApiService.patchSchedule(scheduleId, updatedSchedule);
      const updatedScheduleList = this.state.scheduleList.map((schedule) => {
        return schedule.id === scheduleId ? updatedSchedule : schedule;
      });
      this.setState({
        scheduleList: updatedScheduleList,
      });
    },
    addSchedule: (schedule) => {
      ScheduleApiService.postSchedule(schedule);
      this.state.scheduleList.push(schedule);
    },
    addCurrentSchedule: (currentSchedule) => {
      this.setState({ currentSchedule });
    },
    isSignedIn: TokenService.hasAuthToken(),
    updateSignedIn: (status) => {
      this.setState({ isSignedIn: status });
    },
    error: null,
  };

  componentDidMount() {
    IdleService.setIdleCallback(this.logoutFromIdle);
    if (TokenService.hasAuthToken()) {
      IdleService.regiserIdleTimerResets();

      TokenService.queueCallbackBeforeExpiry(() => {
        AuthApiService.postRefreshToken();
      });
    }
  }

  componentWillUnmount() {
    IdleService.unRegisterIdleResets();
    TokenService.clearCallbackBeforeExpiry();
  }

  logoutFromIdle = () => {
    TokenService.clearAuthToken();
    TokenService.clearCallbackBeforeExpiry();
    IdleService.unRegisterIdleResets();
    this.forceUpdate();
  };

  render() {
    return (
      <div className="App">
        <header className="App__header">
          <Nav />
        </header>
        <TimespaceContext.Provider value={this.state}>
          <main className="App__main">
            <Switch>
              <Route exact path={"/"} component={LandingPage} />
              <Route path={"/signup"} component={SignUpForm} />
              <Route path={"/login"} component={LoginPage} />
              <Route path={"/:name/new-appt"} component={NewApptForm} />
              <PrivateRoute path={"/schedules/:url"} component={Appointments} />
              <PrivateRoute
                path={"/new-schedule"}
                component={NewScheduleForm}
              />
              <PrivateRoute path={"/schedules"} component={Schedules} />
              <Route path={"/success"} component={Success} />
              <Route component={NotFound} />
            </Switch>
          </main>
        </TimespaceContext.Provider>
      </div>
    );
  }
}

export default App;
