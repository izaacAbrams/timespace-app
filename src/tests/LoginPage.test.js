import React from "react";
import ReactDOM from "react-dom";
import LoginPage from "../routes/LoginPage/LoginPage.js";
import renderer from "react-test-renderer";
import App from "../components/App/App";
import { BrowserRouter } from "react-router-dom";

describe("login page route", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <App>
          <LoginPage />
        </App>
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <App>
            <LoginPage />
          </App>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
