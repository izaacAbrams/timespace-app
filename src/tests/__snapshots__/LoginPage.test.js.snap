// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login page route renders UI as expected 1`] = `
<div
  className="App"
>
  <header
    className="App__header"
  >
    <nav
      className="Nav"
    >
      <a
        className="Nav__section"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="Nav__section"
        href="/schedules"
        onClick={[Function]}
      >
        Schedules
      </a>
      <div
        className="Header__not-logged-in"
      >
        <div
          className="Nav__section"
        >
          <a
            className="register"
            href="/signup"
            onClick={[Function]}
          >
            Register
          </a>
          <a
            className="login"
            href="/login"
            onClick={[Function]}
          >
            Log in
          </a>
        </div>
      </div>
    </nav>
  </header>
  <main
    className="App__main"
  >
    <div
      className="LandingPage"
    >
      <h1
        className="LandingPage__title"
      >
        TimeSpaces
      </h1>
      <p
        className="LandingPage__desc"
      >
        TimeSpaces is the ideal tool for small business needs. It is easy to set up and takes all the work out of putting together your schedule! 
        <br />
        <br />
        When you sign up for an account, you will have access to creating your own custom schedules, each with a unique URL to share to anyone wanting to schedule an appointment. 
        <br />
        <br />
        In the case you just want to manually enter in an appointment, not to worry, its easy to create a new appointment or edit an existing one!
      </p>
      <div
        className="LandingPage__btn_wrapper"
      >
        <a
          className="LandingPage__button"
          href="/signup"
          onClick={[Function]}
        >
          Get Started
        </a>
        <a
          className="LandingPage__button"
          href="/sallys-salon-spa/new-appt"
          onClick={[Function]}
        >
          Demo New Appointment
        </a>
      </div>
    </div>
  </main>
</div>
`;
